src/components/subscriptions/licenses/LicenseManagementTable/LicenseManagementRevokeModal.jsx

```diff
@@ -0,0 +1,176 @@
+  const title = `Revoke License${numberToRevoke > 1 ? 's' : ''}`;
+
+  const handleSubmit = () => {
+    setRequestState({ ...initialRequestState, loading: true });
+
+    const userEmailsToRevoke = usersToRevoke.map((user) => user.email);
+    const options = { user_emails: userEmailsToRevoke };
+    LicenseManagerApiService.licenseBulkRevoke(subscription.uuid, options)
+      .then((response) => {
+        setRequestState(initialRequestState);
+        onSuccess(response);
+      })
+      .catch((error) => {
+        logError(error);
+        setRequestState({ ...initialRequestState, error });
+      });
+  };
+
+  const handleClose = () => {
+    if (!requestState.loading) {
+      onClose();
+    }
+  };
+
+  const getRevokeButtonState = () => {
+    if (requestState.error) {
+      return 'error';
+    }
+    if (requestState.loading) {
+      return 'pending';
+    }
+    return 'default';
+  };
+
+  return (
+    <ModalDialog
+      title={title}
+      isOpen={isOpen}
+      onClose={handleClose}
+      hasCloseButton={false}
+    >
+      <ModalDialog.Header>
+        <ModalDialog.Title>
+          {title}
+        </ModalDialog.Title>
+      </ModalDialog.Header>
+      <ModalDialog.Body>
+        {requestState.error
+            && (
+            <Alert variant="danger">
+              <p>There was an error with your request. Please try again.</p>
+              <p>
+                If the error persists,&nbsp;
+                <Hyperlink destination={configuration.ENTERPRISE_SUPPORT_URL}>
+                  contact customer support.
+                </Hyperlink>
+              </p>
+            </Alert>
+            )}
+        {showRevocationCapAlert(subscription.isRevocationCapEnabled, subscription.revocations)
+            && (
+            <Alert variant="warning">
+              You have already revoked {subscription.revocations.applied} licenses. You
+              have {subscription.revocations.remaining} revocations left on your plan.
+            </Alert>
+            )}
+        <p>This action cannot be undone. Learners with revoked licenses must be reinvited. </p>
+        <p>
+          <Hyperlink
+            variant="muted"
+            destination={configuration.ENTERPRISE_SUPPORT_REVOKE_LICENSE_URL}
+          >
+            Learn more about revoking subscription licenses.
+          </Hyperlink>
+        </p>
+      </ModalDialog.Body>
+      <ModalDialog.Footer>
+        <ActionRow>
+          <ModalDialog.CloseButton variant="tertiary">
+            Cancel
+          </ModalDialog.CloseButton>
+          <StatefulButton
+            state={getRevokeButtonState()}
+            variant="danger"
+            onClick={handleSubmit}
+            {...{
+              labels: {
+                default: `Revoke (${numberToRevoke})`,
+                pending: `Revoking (${numberToRevoke})`,
+                complete: 'Done',
+                error: `Retry Revoke (${numberToRevoke})`,
```

Comment:
nit: "Retry revoke" (lowercase "revoke") since only the first letter of button CTAs should be capitalized.